sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -914,10 +914,10 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
       // TODO:Further investigation needed as the CASE translation does not work in orderby queries.
       paginatedProgramQuery.orderBy(
         `CASE           
-          WHEN programs.programStatus = '${ProgramStatus.Pending}' and programs.isActive = true THEN ${SortPriority.Priority1}
-          WHEN programs.programStatus = '${ProgramStatus.Approved}' and programs.isActive = true THEN ${SortPriority.Priority2}
-          WHEN programs.programStatus = '${ProgramStatus.Declined}' and programs.isActive = true THEN ${SortPriority.Priority3}
-          WHEN programs.isActive = false THEN ${SortPriority.Priority4}
+          WHEN programs.programStatus = '${ProgramStatus.Pending}' and programs.isActive = true and (programs.effectiveEndDate is null OR programs.effectiveEndDate > CURRENT_DATE) THEN ${SortPriority.Priority1}
```

Comment:
If the program's effective end date is today's date, then is the program valid till today or has it expired starting today?

Reply:
This is the logic to be considered active: `effectiveEndDate > CURRENT_DATE`.

Reply:
Sounds good. Thanks.